src/System.Net.Http/tests/UnitTests/HttpSystemProxyTest.cs

```diff
@@ -33,12 +33,86 @@ public void HttpProxy_SystemProxy_Loaded()
             Assert.False(HttpSystemProxy.TryCreate(out p));
 
             FakeRegistry.WinInetProxySettings.Proxy = FakeProxyString;
-            WinInetProxyHelper proxyHelper = new WinInetProxyHelper();
 
             Assert.True(HttpSystemProxy.TryCreate(out p));
             Assert.NotNull(p);
             Assert.Equal(fakeProxyUri, p.GetProxy(fooHttp));
             Assert.Equal(fakeProxyUri, p.GetProxy(fooHttps));
         }
+
+        [Theory]
+        [InlineData("http://localhost/", true)]
+        [InlineData("http://127.0.0.1/", true)]
+        [InlineData("http://128.0.0.1/", false)]
+        [InlineData("http://[::1]/", true)]
+        [InlineData("http://foo/", true)]
+        [InlineData("http://www.foo.com/", true)]
+        [InlineData("http://WWW.FOO.COM/", true)]
+        [InlineData("http://foo.com/", false)]
+        [InlineData("http://bar.com/", true)]
+        [InlineData("http://BAR.COM/", true)]
+        [InlineData("http://162.1.1.1/", true)]
+        [InlineData("http://[2a01:5b40:0:248::52]/", false)]
+        [InlineData("http://[2002::11]/", true)]
+        [InlineData("http://[2607:f8b0:4005:80a::200e]/", true)]
+        [InlineData("http://[2607:f8B0:4005:80A::200E]/", true)]
+        [InlineData("http://b\u00e9b\u00e9.eu/", true)]
+        [InlineData("http://www.b\u00e9b\u00e9.eu/", true)]
+        public void HttpProxy_Local_Bypassed(string name, bool shouldBypass)
+        {
+            RemoteInvoke((url, expected) =>
+            {
+                bool expectedResult = Boolean.Parse(expected);
+                IWebProxy p;
+
+                FakeRegistry.Reset();
+                FakeRegistry.WinInetProxySettings.Proxy = FakeProxyString;
+                FakeRegistry.WinInetProxySettings.ProxyBypass = "23.23.86.44;*.foo.com;<local>;BAR.COM; ; 162*;[2002::11];[*:f8b0:4005:80a::200e]; http://www.xn--mnchhausen-9db.at;http://*.xn--bb-bjab.eu;http://xn--bb-bjab.eu;";
+
+                Assert.True(HttpSystemProxy.TryCreate(out p));
+                Assert.NotNull(p);
+
+                Uri u = new Uri(url);
+                Assert.Equal(expectedResult, p.GetProxy(u) == null);
+
+                return SuccessExitCode;
+           }, name, shouldBypass.ToString()).Dispose();
+        }
+
+        [Theory]
+        [InlineData("", 0)]
+        [InlineData(" ", 0)]
+        [InlineData(" ; ;  ", 0)]
+        [InlineData("http://127.0.0.1/", 1)]
+        [InlineData("[::]", 1)]
+        public void HttpProxy_Local_Parsing(string bypass, int count)
+        {
+            RemoteInvoke((bypassValue, expected) =>
```

Comment:
Why is RemoteInvoke being used here?  In most cases, it should only be used if there is real machine state being modified and so the test would be run in separate process.  But with this unit tests using a fake registry, it doesn't seem like it would be necessary.  In general, RemoteInvoke makes the tests take longer to run and is harder to debug test failures.

Reply:
yes, I think it is global for all tests:
System.Net.Http.WinHttpHandler/tests/UnitTests/FakeRegistry.cs
```c#
namespace System.Net.Http.WinHttpHandlerUnitTests
{
    public static class FakeRegistry
    {
...
```